HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+        std::cout << (rotationMatrixError.cwiseQuotient(rotationMatrix1)) * 100 << std::endl;
+
+        std::cout << "\nRotation difference introduced by transformation" << std::endl;
+        const auto differenceRotationMatrix = rotationMatrix1 * rotationMatrix2.inverse();
+        std::cout << differenceRotationMatrix << std::endl;
+
+        std::cout << "\nRotation distance introduced by transformation" << std::endl;
+        const auto axisAngle = Eigen::AngleAxisd(differenceRotationMatrix);
+        auto rotationDistance = axisAngle.angle();
+        std::cout << rotationDistance << std::endl;
+
+        const auto quaternion1 = Eigen::Quaterniond(rotationMatrix1);
+        const auto quaternion2 = Eigen::Quaterniond(rotationMatrix2);
+        rotationDistance = 2 * acos((quaternion1 * quaternion2.conjugate()).w());
+        std::cout << rotationDistance << std::endl;
+
+        return 1;
+    }
+
+    int compareRotationExtended(const Eigen::Array3d rollPitchYaw,
+                                const Eigen::Matrix3d rotationMatrix,
+                                const TransformUtils::Rotation rotation)
+    {
+        const auto rollPitchYawNew = TransformUtils::rotmat2rpy(rotationMatrix, rotation);
+
+        std::cout << "\nOriginal RollPitchYaw:\n" << rollPitchYaw << std::endl;
+        std::cout << "\nNew RollPitchYaw:\n" << rollPitchYawNew << std::endl;
+
+        const auto rotationMatrixNew = TransformUtils::rpy2rotmat(rollPitchYawNew, rotation);
+
+        std::cout << "Original Rotation Matrix:\n" << rotationMatrix << std::endl;
+        std::cout << "New Rotation Matrix:\n" << rotationMatrixNew << std::endl;
+
+        // Calculate error in transformation (Rotation Matrix --> Roll-Pitch-Yaw --> Rotation Matrix)
+        std::cout << "\nVerifying Rotation Matrix" << std::endl;
+        compareRotation(rotationMatrix, rotationMatrixNew);
+
+        return 1;
+    }
+
+} // namespace TransformUtils
+
+int main()
+{
+    try
+    {
+        std::cout << "Opening .YAML file which contains transformation matrix (PoseState node)" << std::endl;
+        auto fileStorage = cv::FileStorage::FileStorage();
+        if(!fileStorage.open("..\\robotTransform.yaml", cv::FileStorage::Mode::READ))
+        {
+            std::cout << "Could not open file" << std::endl;
+        }
+        const auto poseStateNode = fileStorage["PoseState"];
+        std::cout << "Getting PoseState node" << std::endl;
+        if(poseStateNode.empty())
+        {
+            throw "PoseState node not found in file";
+        }
+        const auto poseState = poseStateNode.mat();
+        std::cout << poseState << std::endl;
+        fileStorage.release();
+
+        // Saving all conversions to .YAML file
+        if(!fileStorage.open("robotTransforms.yaml", cv::FileStorage::Mode::WRITE))
+        {
+            std::cout << "Could not open file for writing" << std::endl;
+        }
+        fileStorage.write("PoseState", poseState);
```

Comment:
![image](https://user-images.githubusercontent.com/46567886/69253394-e1e32500-0bb4-11ea-8900-5cba8bb917a8.png)
Can we print it with fewer digits?